api/v1beta1/operatorpolicy_types.go

```diff
@@ -97,30 +64,66 @@ type StatusConfig struct {
-	// only the target namespace will be used and the label selector will be omitted.
+	ComplianceType    policyv1.ComplianceType    `json:"complianceType"`              // musthave
+
+	// Include the name, namespace, and any `spec` fields for the OperatorGroup.
+	// For more info, see `kubectl explain operatorgroup.spec` or
+	// https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/
+	// +kubebuilder:pruning:PreserveUnknownFields
 	// +optional
-	OperatorGroup *OperatorGroup `json:"operatorGroup,omitempty"`
-	// Subscription defines an Application that can be installed
+	OperatorGroup *runtime.RawExtension `json:"operatorGroup,omitempty"`
+
+	// Include the namespace, and any `spec` fields for the Subscription.
+	// For more info, see `kubectl explain subscription.spec` or
+	// https://olm.operatorframework.io/docs/concepts/crds/subscription/
 	// +kubebuilder:validation:Required
-	Subscription SubscriptionSpec `json:"subscription"`
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Subscription runtime.RawExtension `json:"subscription"`
+
 	// Versions is a list of nonempty strings that specifies which installed versions are compliant when
 	// in 'inform' mode, and which installPlans are approved when in 'enforce' mode
-	Versions        []policyv1.NonEmptyString `json:"versions,omitempty"`
-	RemovalBehavior RemovalBehavior           `json:"removalBehavior,omitempty"`
-	StatusConfig    StatusConfig              `json:"statusConfig,omitempty"`
+	Versions []policyv1.NonEmptyString `json:"versions,omitempty"`
+
+	// FUTURE
+	//nolint:dupword
+	// RemovalBehavior RemovalBehavior           `json:"removalBehavior,omitempty"`
+	//nolint:dupword
+	// StatusConfig    StatusConfig              `json:"statusConfig,omitempty"`
 }
 
 // OperatorPolicyStatus defines the observed state of OperatorPolicy
 type OperatorPolicyStatus struct {
 	// Most recent compliance state of the policy
 	ComplianceState policyv1.ComplianceState `json:"compliant,omitempty"`
 	// Historic details on the condition of the policy
-	Condition []metav1.Condition `json:"conditions,omitempty"`
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
 	// List of resources processed by the policy
-	RelatedObject policyv1.RelatedObject `json:"relatedObject"`
+	RelatedObjects []policyv1.RelatedObject `json:"relatedObjects"`
+}
+
+func (status OperatorPolicyStatus) RelatedObjsOfKind(kind string) map[int]policyv1.RelatedObject {
+	objs := make(map[int]policyv1.RelatedObject)
+
+	for i, related := range status.RelatedObjects {
+		if related.Object.Kind == kind {
+			objs[i] = related
+		}
+	}
+
+	return objs
+}
+
+// Searches the conditions of the policy, and returns the index and condition matching the
+// given condition Type. It will return -1 as the index if no condition of the specified
+// Type is found.
+func (status OperatorPolicyStatus) GetCondition(condType string) (int, metav1.Condition) {
+	for i, cond := range status.Conditions {
+		if cond.Type == condType {
```

Comment:
What if multiple conditions match? is it okay to return only the first match?

Reply:
I think it's ok to just return the first match. Ideally we would have some way of enforcing that there are not multiple conditions of the same type (the actual k8s APIs can use listType=map [like this](https://github.com/kubernetes/apimachinery/blob/v0.27.7/pkg/apis/meta/v1/types.go#L1475), but that didn't seem to be available for CRDs). We'll just need to be careful not to accidentally put the same condition twice.

Reply:
good to learn Thanks